aptos-move/aptos-gas-meter/src/algebra.rs

```diff
@@ -156,10 +208,17 @@ impl GasAlgebra for StandardGasAlgebra {
             },
         );
 
-        self.charge(gas_consumed_internal)?;
+        let (actual, res) = self.charge(gas_consumed_internal);
+        if self.feature_version >= 12 {
+            self.storage_fee_in_internal_units += actual;
+            self.storage_fee_used += amount;
```

Comment:
We don't care to proportionate the `amount`? 😂

Reply:
Good point. The real amount being charged is one that first gets converted into gas units and then converted back. Do you think it makes more sense for us to record that instead?

Reply:
Yeah this field is not that important -- can't be 100% precise anyway. Just wanted to make the FeeStatement makes a bit more sense when we hit OUT_OF_GAS in the middle of charging it. Again, seems no perfect solution here since the converted value has to be part of the total gas used.